.github/workflows/docs-preview.yml

```diff
@@ -0,0 +1,140 @@
+name: Docs Preview
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.run_id }}
+  cancel-in-progress: true
+
+on:
+  pull_request_target:
+    branches:
+      - main
+
+jobs:
+  deploy_preview_docs:
+    name: deploy-preview-docs
+    runs-on: ubuntu-latest
+    needs:
+      - should_run_docs
+    steps:
+      - name: Create initial PR comment
+        if: ${{ needs.should_run_docs.outputs.should_run_docs == 'true' }}
+        uses: marocchino/sticky-pull-request-comment@v2
```

Comment:
**issue (blocking)** Our security policy does not permit the execution of actions by unverified creators.